#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.9999564619581492 0.003218185467974895 0.00875885097681851 1.1481159287927065
  position 0.020814744882016567 2.2541698207008873 1.035065074757991
}
TexturedBackground {
  texture "factory"
}
TexturedBackgroundLight {
}
DEF area RectangleArena {
  floorSize 1.016 1.016
  floorTileSize 0.508 0.508
  wallAppearance BrushedAluminium {
    colorOverride 0.4 0.4 0.4
  }
}
DEF epuck E-puck {
  translation 0.365566 1.33227e-15 0.392146
  rotation -0.6979647042481143 -0.030090787249488382 -0.715499696817929 0.0005653071795675667
  controller "motionEstimationNoise"
  supervisor TRUE
  camera_width 50
  turretSlot [
    InertialUnit {
      rotation 0 1 0 3.14
      noise 0.25
    }
    DEF righthinge HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
          controlPID 8 2 0.5
        }
        PositionSensor {
          name "right wheel sensor"
          noise 0.25
        }
      ]
      endPoint Solid {
        translation 0.04500995778380096 0.025013380758883635 -2.4671929852070174e-05
        rotation 0.9999993203499804 0.0010052734270078172 -0.0005905293508524928 4.6368371220546685
        children [
          DEF WHEEL Transform {
            translation -0.007 0.01 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "right wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
        rotationStep 0.25
      }
    }
    DEF lefthinge HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
          controlPID 8 0 0
        }
        PositionSensor {
          name "left wheel sensor"
          noise 0.25
        }
      ]
      endPoint Solid {
        translation -0.04501012942703836 0.024977671977279067 -4.5733976162035025e-05
        rotation -0.9999982113395169 0.00047275268613733136 -0.0018313444964861226 1.2425810202889593
        children [
          DEF WHEEL Transform {
            translation 0.007 0 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.1 0.1 0.1
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "left wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    DEF front_ds DistanceSensor {
      translation 0 0 -0.045
      rotation 0 1 0 1.5708
      name "front_ds"
      recognitionColors [
        0 0 0
      ]
      lookupTable [
        0 0 0
        1.27 1.27 0.25
      ]
    }
    DEF right_ds DistanceSensor {
      translation 0.035 -0.01 -0.01
      name "right_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0.25
      ]
    }
    DEF left_ds DistanceSensor {
      translation -0.035 -0.01 -0.01
      rotation 0 1 0 3.14159
      name "left_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0.25
      ]
    }
    Camera {
      translation 0 -0.01 -0.045
      name "camera1"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
  ]
}
